test/libsolidity/semanticTests/externalSource/non_normalized_paths.sol

```diff
@@ -0,0 +1,28 @@
+==== ExternalSource: _non_normalized_paths/a.sol ====
+==== ExternalSource: _non_normalized_paths//b.sol ====
+==== ExternalSource: C/////c.sol=_non_normalized_paths/c.sol ====
+==== ExternalSource: C/../////D/d.sol=_non_normalized_paths///d.sol ====
+import "_non_normalized_paths/a.sol";
+import "_non_normalized_paths/b.sol";
+import "C/c.sol";
+import "D/d.sol";
+
+contract Contract {
+    A _a;
+    B _b;
+    C _c;
+    D _d;
+    constructor() {
+        _a = new A();
+        _b = new B();
+        _c = new C();
+        _d = new D();
+    }
+    function foo() public returns (uint) {
+        return _a.foo() + _b.foo() + _c.foo() + _d.foo();
+    }
+}
+// ====
+// compileViaYul: also
+// ----
+// foo() -> 0x04f5
```

Comment:
As discussed on the channel, these tests turned out to be quite verbose and we should try to trim them down a bit. Especially for the ones I suggested in https://github.com/ethereum/solidity/pull/11110#discussion_r617074805 the only relevant things are the paths and imports. You could safely remove almost all functions and these tests will provide the same value. As long as it compiles, it's good so the semantic part can be minimal.

For example in this one I would just use an empty `contract Contract {}`.

In `a.sol` I would only leave `contract A {}` and then import it with
```solidity
import {A} from "_non_normalized_paths//a.sol";
```
This is enough to make the compiler fail if it imports the wrong file and `A` is not available in it.

Reply:
Ok, I simplified this test a bit.

Reply:
Looks good. Could you simplify `relative_imports.sol`, `source_name_starting_with_dots.sol`, `_relative_imports/dir/contract.sol` and other files from the `_` dirs that way too? I think there's still a lot that can be cut.